tests/PHPStan/Analyser/data/bug-8775.php

```diff
@@ -0,0 +1,279 @@
+ * '80': int,
+ * '81': int,
+ * '82': int,
+ * '83': int,
+ * '84': int,
+ * '85': int,
+ * '86': int,
+ * '87': int,
+ * '88': int,
+ * '89': int,
+ * '90': int,
+ * '91': int,
+ * '92': int,
+ * '93': int,
+ * '94': int,
+ * '95': int,
+ * '96': int,
+ * '97': int,
+ * '98': int,
+ * '99': int,
+ * '100': int,
+ * '101': int,
+ * '102': int,
+ * '103': int,
+ * '104': int,
+ * '105': int,
+ * '106': int,
+ * '107': int,
+ * '108': int,
+ * '109': int,
+ * '110': int,
+ * '111': int,
+ * '112': int,
+ * '113': int,
+ * '114': int,
+ * '115': int,
+ * '116': int,
+ * '117': int,
+ * '118': int,
+ * '119': int,
+ * '120': int,
+ * '121': int,
+ * '122': int,
+ * '123': int,
+ * '124': int,
+ * '125': int,
+ * '126': int,
+ * '127': int,
+ * '128': int,
+ * '129': int,
+ * '130': int,
+ * '131': int,
+ * '132': int,
+ * '133': int,
+ * '134': int,
+ * '135': int,
+ * '136': int,
+ * '137': int,
+ * '138': int,
+ * '139': int,
+ * '140': int,
+ * '141': int,
+ * '142': int,
+ * '143': int,
+ * '144': int,
+ * '145': int,
+ * '146': int,
+ * '147': int,
+ * '148': int,
+ * '149': int,
+ * '150': int,
+ * '151': int,
+ * '152': int,
+ * '153': int,
+ * '154': int,
+ * '155': int,
+ * '156': int,
+ * '157': int,
+ * '158': int,
+ * '159': int,
+ * '160': int,
+ * '161': int,
+ * '162': int,
+ * '163': int,
+ * '164': int,
+ * '165': int,
+ * '166': int,
+ * '167': int,
+ * '168': int,
+ * '169': int,
+ * '170': int,
+ * '171': int,
+ * '172': int,
+ * '173': int,
+ * '174': int,
+ * '175': int,
+ * '176': int,
+ * '177': int,
+ * '178': int,
+ * '179': int,
+ * '180': int,
+ * '181': int,
+ * '182': int,
+ * '183': int,
+ * '184': int,
+ * '185': int,
+ * '186': int,
+ * '187': int,
+ * '188': int,
+ * '189': int,
+ * '190': int,
+ * '191': int,
+ * '192': int,
+ * '193': int,
+ * '194': int,
+ * '195': int,
+ * '196': int,
+ * '197': int,
+ * '198': int,
+ * '199': int,
+ * '200': int,
+ * '201': int,
+ * '202': int,
+ * '203': int,
+ * '204': int,
+ * '205': int,
+ * '206': int,
+ * '207': int,
+ * '208': int,
+ * '209': int,
+ * '210': int,
+ * '211': int,
+ * '212': int,
+ * '213': int,
+ * '214': int,
+ * '215': int,
+ * '216': int,
+ * '217': int,
+ * '218': int,
+ * '219': int,
+ * '220': int,
+ * '221': int,
+ * '222': int,
+ * '223': int,
+ * '224': int,
+ * '225': int,
+ * '226': int,
+ * '227': int,
+ * '228': int,
+ * '229': int,
+ * '230': int,
+ * '231': int,
+ * '232': int,
+ * '233': int,
+ * '234': int,
+ * '235': int,
+ * '236': int,
+ * '237': int,
+ * '238': int,
+ * '239': int,
+ * '240': int,
+ * '241': int,
+ * '242': int,
+ * '243': int,
+ * '244': int,
+ * '245': int,
+ * '246': int,
+ * '247': int,
+ * '248': int,
+ * '249': int,
+ * '250': int,
+ * '251': int,
+ * '252': int,
+ * '253': int,
+ * '254': int,
+ * '255': int,
+ * '256': int,
+ * '257': int,
+ * }
+ */
+class Test
+{
+	/**
+	 * @var Fixtures
+	 */
+	public static $fixtures;
+
+	function doFoo():void {
+		assertType('(int|string)', Test::$fixtures['257']);
```

Comment:
this appears to be the type before this fix? so the fix isn't actually tested here?
https://phpstan.org/r/2e0b1bf7-c51c-42ff-b6f4-eba9e384506c

Reply:
so is there no way for this to be marked as just an int then? 256 is a hard limit for type inference?

Reply:
Before the fix the type was `int|string`. Now it is `(int|string)` which means its benevolent (more forgiving).

For arrays bigger then 256 elements we are not that precise anymore for performance reasons

Reply:
thanks, i didn't know what benevolent union was!